plugins/com.google.cloud.tools.eclipse.ui.test/src/com/google/cloud/tools/eclipse/ui/privacytos/PrivacyTosStartupTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.ui.privacytos;
+
+import static org.mockito.Mockito.verify;
+
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.jface.dialogs.MessageDialogWithToggle;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.osgi.service.prefs.Preferences;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class PrivacyTosStartupTest {
+  @Mock private PrivacyTosStartup ptStartup;
+  
+  public static void configureShowAgain(boolean showAgain) {
+    IEclipsePreferences preferenceStore =
+        InstanceScope.INSTANCE.getNode(PrivacyTosStartup.PREFERENCE_NODE_KEY);
+    Preferences preferences = preferenceStore.node(PrivacyTosStartup.PREFERENCE_NODE_KEY);
+    preferences.put(PrivacyTosStartup.PREFERENCE_PERSISTENCE_KEY, showAgain ? MessageDialogWithToggle.PROMPT : MessageDialogWithToggle.NEVER);
+  }
+  
+  @BeforeClass
+  public static void setup() {
+    configureShowAgain(false);
+  }
+  
+  @Test
+  public void testRunAtStartup() {
+    configureShowAgain(true);
+    verify(ptStartup).earlyStartup();
+  }
```

Comment:
Does it makes sense to also have a test verifying that if the "NEVER" option is in effect, then the startup dialog does not show up? Or is it hard to prove a negative with UI tests?

Reply:
I would have to research on how to check the state of a component/dialog within eclipse, I'll try to find something similar in other tests

Reply:
It would be something like `verify(ptStartup, times(0)).earlyStartup();`, but it's also unimportant. Ignore this comment, then.